future_integration_tests/issue_3593.rs

```diff
@@ -0,0 +1,24 @@
+use winit::event_loop::EventLoop;
+use winit::window::Window;
+use rwh_06::HasWindowHandle;
+
+fn issue_3593() {
+    // We should not be able to get a window handle off of the main thread.
+    let event_loop = EventLoop::new().unwrap();
+    event_loop.run(|_, elwt| {
```

Comment:
Don't use deprecated API.

Reply:
Is there a better way to create a closure application ad-hoc?

Reply:
Not really, no. This would have to be done with `ApplicationHandler`:
```rust
struct App;
impl ApplicationHandler for App {
    fn new_events(&mut self, event_loop: &ActiveEventLoop, _cause: StartCause) {
        // Test code in here
    }
}

fn issue_3593() {
    let event_loop = EventLoop::new().unwrap();
    event_loop.run_app(App).unwrap();
}
```

I guess for the purpose of integration tests, we should probably have some sort of mechanism to do this more easily.